tox.ini

```diff
@@ -4,20 +4,44 @@
 # and then run "tox" from this directory.
 
 [tox]
-envlist = py35,py36,py37,py38,pypy3,flake8
+envlist = flake8,packaging,docs,py{35,36,37,38},pypy3
 
 [testenv]
-basepython =
-    py35: python3.5
-    py36: python3.6
-    py37: python3.7
-    py38: python3.8
-    pypy3: pypy3
-
-commands = {envpython} setup.py test
+deps =
+    pytest
+    pytest-tldr
+whitelist_externals =
+    make
+commands =
+    make
+    pytest -vv
 
 [testenv:flake8]
 basepython = python3.8
 deps =
     flake8
 commands = flake8 {posargs}
+
+[testenv:package]
+basepython = python3.8
+deps =
+    wheel
+    twine
+commands =
+    python setup.py sdist bdist_wheel
+    python -m twine check dist/*
+
+[testenv:docs]
```

Comment:
Minor thing: I would reorder the section here so that the `docs` environment isn't in between the `package` and `publish` environments - the last two environments are closely related, so it makes sense to have them defined together.